designs/provisioner-priority.md

```diff
@@ -0,0 +1,196 @@
+# Prioritizing Provisioners
+
+## Goals
+
+- Allowing a user to describe a logical ordering to their provisioners
+- Allowing a user to set defaults and/or fallback orderings for their provisioners
+
+## Use Cases
+
+
+1. Creating a strongly preferred default provisioner that will always be attempted first, except when more specific configuration is required. This enables scenarios like specifying `kubernetes.io/arch=arm64` for workloads that require this architecture, while preferring `kubernetes.io/arch=amd64` on all other workloads, even if the pod does not contain specific `nodeSelector` constraints. This is particularly useful for those that have high numbers of workloads that only work on `amd64` instances but they do not want to have to go and assign the specific architecture-specific `nodeSelector` to each of these workloads.
+
+   **Example**
+
+    ```yaml
+    # Default provisioner that will be attempted to be scheduled first
+
+    apiVersion: karpenter.sh/v1alpha5
+    kind: Provisioner
+    metadata:
+      name: default
+    spec:
+      weight: 100
+      requirements:
+      - key: kubernetes.io/arch
+        operator: In
+        values: ["amd64"]
+    ```
+
+    ```yaml
+    # ARM-64 specific provisioner that will be scheduled if the other provisioner does not fit the constraints for this provisioner
+
+    apiVersion: karpenter.sh/v1alpha5
+    kind: Provisioner
+    metadata:
+      name: arm64
+    spec:
+      weight: 50
+      requirements:
+      - key: kubernetes.io/arch
+        operator: In
+        values: ["arm64"]
+    ```
+
+2. Creating a strongly preferred default provisioner for enterprise users that have `reserved` instance types and want to prefer having these instance types scheduled prior to other instance types. This assumes that we eventually support the `karpenter.sh/capacity-type=reserved` label.
```

Comment:
we don't need to assume the reserved label. It would be reasonable for users to just fill in the instance types that they have purchases RIs for in the provisioner.

Reply:
Ah, got it makes sense. My assumption is that these instance types fall under the `on-demand` capacity type

Reply:
yes, RIs are a billing concept, not a capacity type, but they do map to on-demand instances.